### PR TITLE
Skipping Admin-only tests instead of failing them

### PR DESCRIPTION
**Pull request by agr**
Got fed up with all the redness in test explorer. This introduces two additional test attributes that will automagically skip if tests are executed not as admin. Since there were two test assemblies with admin-only tests, had to introduce shared test library with common code.

Tested in VS: when run as regular user, skips all tests that failed before, when run as admin runs them fine. Tested that CI still runs those tests after the change.

To enable test skip one needs to add the `ENABLE_NONADMIN_TEST_SKIP` environment variable. This was done to reduce the chance of accidentally skipping those tests in CI (in initial implementation they would be skipped if CI started to run tests as non-admin).

### CODE REVIEW

File: NuGet.Jobs.sln [hunk 1]
```diff
@@ -147,6 +147,8 @@ Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Tests.Stats.CDNLogsSanitize
 EndProject
 Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Monitoring.PackageLag.Tests", "tests\Monitoring.PackageLag.Tests\Monitoring.PackageLag.Tests.csproj", "{D3F1711A-25AC-4EC9-9971-4F838BCD2A07}"
 EndProject
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "TestUtil", "tests\TestUtil\TestUtil.csproj", "{C3F84BAD-ACFA-4AE3-8286-D12F5A5BBC62}"
```
Comment (agr) [inline NuGet.Jobs.sln]: Need to update build scripts as well.

File: tests/TestUtil/UserHelper.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace TestUtil
+{
+    public static class UserHelper
+    {
+        public static bool IsAdmin()
+        {
+            var identity = WindowsIdentity.GetCurrent();
+            var principal = new WindowsPrincipal(identity);
+            bool isAdmin = principal.IsInRole(WindowsBuiltInRole.Administrator);
```
Comment (loic-sharma) [inline tests/TestUtil/UserHelper.cs]:
```suggestion
            return principal.IsInRole(WindowsBuiltInRole.Administrator);
```

File: tests/TestUtil/FactIfAdminAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+using System;
+using Xunit;
+using Xunit.Sdk;
+
+namespace TestUtil
+{
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
+    [XunitTestCaseDiscoverer("Xunit.Sdk.FactDiscoverer", "xunit.execution.{Platform}")]
+    public class FactIfAdminAttribute : FactAttribute
```
Comment (agr) [inline tests/TestUtil/FactIfAdminAttribute.cs]: Any name suggestions? `AdminOnlyFact` maybe?

Comment (joelverhagen) [reply tests/TestUtil/FactIfAdminAttribute.cs]: I like `AdminOnly`

Comment (agr) [reply tests/TestUtil/FactIfAdminAttribute.cs]: We have theories, too, also `AdminOnly` does not emphasize that it's a test.

**Pull request merged**